src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,150 @@
+                String[] tokens = currTask.split("\\| ");
+                if (tokens[0].charAt(0) == "T".charAt(0)) {
+                    Task task = new Todo("todo " + tokens[2]);
+                    if (Integer.parseInt(tokens[1].trim()) == 1) {
+                        task.toggleStatus();
+                    }
+                    store.add(task);
+                }
+                if (tokens[0].charAt(0) == "D".charAt(0)) {
+                    Task task = new Deadline("deadline " + tokens[2] + "/by " + tokens[3]);
+                    if (Integer.parseInt(tokens[1].trim()) == 1) {
+                        task.toggleStatus();
+                    }
+                    store.add(task);
+                }
+                if (tokens[0].charAt(0) == "E".charAt(0)) {
+                    Task task = new Event("event " + tokens[2] + "/at " + tokens[3]);
+                    if (Integer.parseInt(tokens[1].trim()) == 1) {
+                        task.toggleStatus();
+                    }
+                    store.add(task);
+                }
+            }
+            fr.close();
+        } catch (IOException e) {
+            throw new LoadException();
+        }
+    }
+
+    public TaskList(String filePath) {
+        this.file = new File(filePath);
+        this.store = new ArrayList<>();
+
+        if (!file.getParentFile().exists()) {
+            file.getParentFile().mkdirs();
+        }
+    }
+
+    /**
+     * Retrieves task from given index
+     *
+     * @param index
+     * @return Task in given index
+     */
+    public Task get(int index) {
+        return this.store.get(index);
+    }
+
+    public void printTasks() {
+        int i = 1;
+        for (Task task : this.store) {
+            System.out.print(i + ".");
+            System.out.println(task);
+            i++;
+        }
+    }
+
+    public void findTasks(String substring) {
+        List<Task> tempList = new ArrayList<>();
+        for (Task task : this.store) {
+            if (task.getDescription().contains(substring)) {
+                tempList.add(task);
+            }
+        }
+
+        int i = 1;
+        for (Task task : tempList) {
+            System.out.print(i + ".");
+            System.out.println(task);
+            i++;
+        }
+    }
+
+    /**
+     * Get number of tasks in current tasklist
+     *
+     * @return int value of size of tasklist
+     */
+    public int size() {
+        return this.store.size();
+    }
+
+    /**
+     * Adds param task into tasklist
+     *
+     * @param task
+     */
+    public void add(Task task) {
+        this.store.add(task);
+    }
+
+    /**
+     * Removes task from param index from tasklist
+     *
+     * @param index
+     * @return task removed from tasklist
+     */
+    public Task remove(int index) {
+        return this.store.remove(index);
+    }
+
+    /**
+     * Writes current state snapshot of tasklist into .txt file
+     *
+     * @throws WriteException
```

Comment:
Perhaps you could add a one-liner statement on when will the exception be thrown.